src/server/pkg/deploy/assets/assets.go

```diff
@@ -1205,27 +1206,47 @@ func EtcdNodePortService(local bool, opts *AssetOpts) *v1.Service {
 // EtcdHeadlessService returns a headless etcd service, which is only for DNS
 // resolution.
 func EtcdHeadlessService(opts *AssetOpts) *v1.Service {
+	ports := []v1.ServicePort{
+		{
+			Name: "peer-port",
+			Port: 2380,
+		},
+	}
+	return makeHeadlessService(opts, etcdName, etcdHeadlessServiceName, ports)
+}
+
+// PostgresHeadlessService returns a headless postgres service, which is only for DNS
+// resolution.
+func PostgresHeadlessService(opts *AssetOpts) *v1.Service {
```

Comment:
We talked offline about breaking apart assets.go.  I think a `postgres.go` with this stuff would be a good start.